test/androidTest/java/org/thoughtcrime/securesms/espresso/ViewActions.java

```diff
@@ -69,4 +69,45 @@ public void perform(UiController uiController, View view) {
             }
         };
     }
+
+    public static ViewAction clickParent(final int levels) {
+        return new ViewAction() {
+            @Override
+            public Matcher<View> getConstraints() {
+                return null;
+            }
+
+            @Override
+            public String getDescription() {
+                return "clicking on parent view";
+            }
+
+            @Override
+            public void perform(UiController uiController, View view) {
+                for (int i = 0; i < levels; i++) {
+                    view = (View)view.getParent();
```

Comment:
Space after type casting.

Reply:
I've looked in the codebase and that doesn't seem to be the pattern

Reply:
Ah, it seems that Signal does not respect the official [Oracle Java Style Guide](http://www.oracle.com/technetwork/java/codeconventions-150003.pdf).

Then it's okay.